### DISCUSSION TITLE
Show albums for multiple images

### DISCUSSION THREAD

**Discussion by turt2live** [element-hq/element-meta#754] [Ideas]
### Description

This could either be a new event type or determined on the fly by Riot. The idea would be that if someone was uploading 2 or more images to the room, then the images get put into an album for scrolling through - much like Instagram. The download link should be for the entire set, with per-image download links being accessible somewhere (like the lightbox). 

I'd recommend a new event type of `m.album` which would essentially be an array of `m.image` bodies. However, there's also the possibility of Riot automatically grouping images to albums using some rules:
* Only group images if they are consecutive and from the same user
* Don't group images if they were sent too far apart (~1min?) - group based on previous image
  * Slow uploads are the reason to look at the previous image. For example, if A was uploaded at 00:00:00, B at 00:00:45, and C at 00:01:10 - A, B, and C should be grouped, even though C is >1 min away from A (as it is <1 min away from B).

### Version information

- **Platform**: web (in-browser)
- **Browser**: Chrome 63
- **OS**: Windows 10
- **URL**: riot.im/develop

**Comment by MurzNN** [comment]
Better way is attach multiple images to single message in UI, so when user composing message text - `Upload file` button will upload attached file to current composing message (when it will be sent), instead of immediately send this file as separate message to room. This is like many other messengers works with image attachments.
After implementing this will be much easier to detect albums and make slideshow from images.
Here https://github.com/vector-im/riot-web/issues/6295 is my feature request about this.

**Comment by t3chguy** [comment]
Related to https://github.com/vector-im/riot-web/issues/337

**Comment by kittykat** [comment]
Moving to a meta discussion as this is cross-platform

**Reply by beedell-roke** [reply]
@kittykat, surely, all `meta` issues are cross-platform...?
